core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -742,6 +745,40 @@ object Parser {
         }
       }
 
+    implicit val catsShowExpectation: Show[Expectation] =
+      new Show[Expectation] {
+        private val dq = "\""
+        def show(expectation: Expectation): String = expectation match {
+          case OneOfStr(_, strs: List[String]) =>
+            "one of: " + strs.map(s => dq + s + dq).mkString("{", ", ", "}")
+
+          case InRange(_, lower: Char, upper: Char) =>
+            if (lower != upper) s"in range: ['$lower', '$upper']"
+            else s"is: '$lower'"
+
+          case StartOfString(_) =>
+            "start of string"
+
+          case EndOfString(_, length) =>
+            s"end of string, length = $length"
+
+          case Length(_, expected, actual) =>
+            s"length: expected = $expected, actual = $actual"
+
+          case ExpectedFailureAt(_, matched) =>
+            s"failure at $matched"
```

Comment:
I think this is unclear, I think it should be `expected failure but matched $matched` or something. This expectation means we *didn't* fail when we thought we would. Note, it is confusing that this and `fail` and `fail: ` all start the same but are rather different.